components/org.wso2.carbon.identity.context.rewrite.valve/src/main/java/org/wso2/carbon/identity/context/rewrite/valve/TenantContextRewriteValve.java

```diff
@@ -273,17 +280,47 @@ private void handleInvalidTenantDomainErrorResponse(Response response, int error
         response.getWriter().print(errorResponse.toString());
     }
 
-    private void handleRestrictedTenantDomainErrorResponse(Response response) throws IOException, ServletException {
+    private void handleRestrictedTenantDomainErrorResponse(Request request, Response response) throws IOException {
+
+        String requestContentType = request.getContentType();
+        if (StringUtils.contains(requestContentType, "application/json")) {
+            response.setContentType("application/json");
+            response.setStatus(HttpServletResponse.SC_FORBIDDEN);
+            response.setCharacterEncoding("UTF-8");
+            JsonObject errorResponse = new JsonObject();
+            errorResponse.addProperty("code", HttpServletResponse.SC_FORBIDDEN);
+            String errorMsg = "Access to super tenant domain over tenanted URL format (/t/carbon.super) is restricted. "
+                    + "Please use the server base path instead.";
+            errorResponse.addProperty("message", errorMsg);
+            errorResponse.addProperty("description", errorMsg);
+            response.getWriter().print(errorResponse.toString());
+        } else {
+            response.setContentType("text/html");
+            String errorPage = loadPageNotFoundErrorPage();
+            response.getWriter().print(errorPage);
+        }
+    }
 
-        response.setContentType("application/json");
-        response.setStatus(HttpServletResponse.SC_FORBIDDEN);
-        response.setCharacterEncoding("UTF-8");
-        JsonObject errorResponse = new JsonObject();
-        String errorMsg = "Access to super tenant domain over tenanted URL format (/t/carbon.super) is restricted. " +
-                "Please use the server base path instead.";
-        errorResponse.addProperty("code", HttpServletResponse.SC_FORBIDDEN);
-        errorResponse.addProperty("message", errorMsg);
-        errorResponse.addProperty("description", errorMsg);
-        response.getWriter().print(errorResponse.toString());
+    private String loadPageNotFoundErrorPage() {
+
+        String errorPage = "Page Not Found";
+        try {
+            Path pageNotFoundHtmlResponse =
+                    Paths.get(CarbonUtils.getCarbonHome(), "repository", "resources", "identity", "pages",
+                            "page_not_found.html");
+            if (!Files.exists(pageNotFoundHtmlResponse) ||
+                    !Files.isRegularFile(pageNotFoundHtmlResponse)) {
+                if (log.isDebugEnabled()) {
+                    log.debug("pageNotFoundHtmlResponse is not present at: " + pageNotFoundHtmlResponse);
+                }
+            }
```

Comment:
```suggestion
                }
                return errorPage;
            }
```
Need to break the flow here if the file is not available right?

Reply:
If the file is not available, the default content will be used as the error page content without breaking the flow while logging the warning message.